crates/op-rpc-types/src/op/block.rs

```diff
@@ -0,0 +1,1516 @@
+//! Block RPC types.
+
+#![allow(unknown_lints, non_local_definitions)]
+
+use crate::{other::OtherFields, Transaction, Withdrawal};
+use alloy_eips::{calc_blob_gasprice, calc_excess_blob_gas};
+use alloy_primitives::{
+    ruint::ParseError, Address, BlockHash, BlockNumber, Bloom, Bytes, B256, U256, U64,
+};
+use alloy_rlp::{bytes, Decodable, Encodable, Error as RlpError};
+use serde::{
+    de::{MapAccess, Visitor},
+    ser::{Error, SerializeStruct},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::{collections::BTreeMap, fmt, num::ParseIntError, ops::Deref, str::FromStr};
+
+/// Block representation
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Block {
+    /// Header of the block.
+    #[serde(flatten)]
+    pub header: Header,
+    /// Uncles' hashes.
+    #[serde(default)]
+    pub uncles: Vec<B256>,
+    /// Block Transactions. In the case of an uncle block, this field is not included in RPC
+    /// responses, and when deserialized, it will be set to [BlockTransactions::Uncle].
+    #[serde(
+        default = "BlockTransactions::uncle",
+        skip_serializing_if = "BlockTransactions::is_uncle"
+    )]
+    pub transactions: BlockTransactions,
+    /// Integer the size of this block in bytes.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub size: Option<U256>,
+    /// Withdrawals in the block.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub withdrawals: Option<Vec<Withdrawal>>,
+    /// Support for arbitrary additional fields.
+    #[serde(flatten)]
+    pub other: OtherFields,
+}
+
+impl Block {
+    /// Converts a block with Tx hashes into a full block.
+    pub fn into_full_block(self, txs: Vec<Transaction>) -> Self {
+        Self { transactions: BlockTransactions::Full(txs), ..self }
+    }
+}
+
+/// Block header representation.
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct Header {
```

Comment:
is this the same header as ethereum?

Reply:
Yes. Optimism RPC's return the same data.